src/gui/torrentcontenttreeview.cpp

```diff
@@ -94,7 +94,7 @@ void TorrentContentTreeView::renameSelectedFile(BitTorrent::TorrentHandle *torre
     const QModelIndexList selectedIndexes = selectionModel()->selectedRows(0);
     if (selectedIndexes.size() != 1) return;
 
-    const QModelIndex modelIndex = selectedIndexes.first();
+    const QPersistentModelIndex modelIndex = selectedIndexes.first();
```

Comment:
Is it enough? Can you explain?

Reply:
from https://doc.qt.io/qt-5/qmodelindex.html#details:
>Note: Model indexes should be used immediately and then discarded. You should not rely on indexes to remain valid after calling model functions that change the structure of the model or delete items. If you need to keep a model index over time use a QPersistentModelIndex.

And notice that I added `!modelIndex.isValid()` below.

Reply:
I see it. But what's the real difference between "isValid()" of these two classes? Sorry I cannot to check it myself now.

Reply:
QModelIndex::isValid() seems doesn't do much checking:
https://github.com/qt/qtbase/blob/7e2016de5ee6d9cc2c5d6b94fab4bc56a8a33303/src/corelib/itemmodels/qabstractitemmodel.h#L76

QPersistentModelIndex::isValid() seems to check some kind of stored persistent index:
https://github.com/qt/qtbase/blob/60feaae1967aca5c7b1b0f6cb732962ddd1b2115/src/corelib/itemmodels/qabstractitemmodel.cpp#L465
https://github.com/qt/qtbase/blob/60feaae1967aca5c7b1b0f6cb732962ddd1b2115/src/corelib/itemmodels/qabstractitemmodel.cpp#L61